capstone_project/src/main/java/com/google/sps/servlets/SortServlet.java

```diff
@@ -46,12 +48,16 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     Query query = new Query("Post");
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    List<Entity> posts = datastore.prepare(query).asList(FetchOptions.Builder.withDefaults());
+    List<Entity> postEntities =
+        datastore.prepare(query).asList(FetchOptions.Builder.withDefaults());
+
     PostService postService = PostService.Builder.builder().build();
+    postService.sortEntities(sortType, postEntities);
 
-    postService.sortEntities(sortType, posts);
-    response.setContentType("application/json");
+    List<Post> posts =
+        postEntities.stream().map(PostService::convertEntityToPost).collect(Collectors.toList());
```

Comment:
future work: it seems like you can just combine the logic in this servlet with the PostServlet? "not sorting" could be considered a type of sorting if you squint

Reply:
Yea this makes sense. I'll probably do it in a separate PR as this would require some changes to javascript code, which would be irrelevant to this PR